### DISCUSSION TITLE
Fast "traveling" across app

### DISCUSSION THREAD

**Discussion by Doburo** [wuwanahq/wuwana2#80] [General]
Fast traveling across yours application cause me some questions:
   1) Why you dont use one point to enter all yours application? 
   2) You copy-paste a lot of duplicate things, with same functionality... 
Like:
    
    ```
    spl_autoload_register(function($className) {
	    require 'Models/' . str_replace('\\', '/', $className) . '.php';
    });
    --------------------------------------
    $language = WebApp\WebApp::getLanguageCode();
    
    $companies = WebApp\Data::getCompany()->selectCategoriesRegions($language, [], $selectedRegions, $limit);
    
    require 'Templates/text ' . $language . '.php';
    require 'homepage/text ' . $language . '.php';
    require 'homepage/view.php';
    --------------------------------------
    f (memory_get_peak_usage() > WebApp\WebApp::MEMORY_LIMIT)
    { trigger_error(memory_get_peak_usage() . ' Bytes of memory used', E_USER_NOTICE); }
    ```
3) For company info from Instagram, why not try to use some  API. Instagram already have it, so...
Simple response from Instagram via link: [Wuwana.es Instagram](https://www.instagram.com/wuwana.es/?__a=1)

![image](https://user-images.githubusercontent.com/52839372/103163304-2d7af000-4805-11eb-9943-50df36001710.png)

Well i think its same to Twitter, etc.

4) Well for some reason you use a lot of `trigger_error`, so again, why not use some Exception Handler for app.

I'm not trying to show you how many mistakes there are. It's just that when you expand the application, this will annoy for you a lot. And since you decide to write your own framework, you should pay special attention to the fact that the application must have one initialization point. (can't share values from model to view, a lot of duplicates, hard to use admin sessions, etc) 
The more files will take on the role of managers, the more problematic it is to work with the application (there are no connections, everyone is his own boss).
It will also help you with "communication" between business logic, and your desire to make statistics.

And I'm not even junior in development, so just trying to say my own point of view, based on books / videos about backend programming.

Best wishes from me.

**Comment by Nils85** [comment]
Hey @Doburo !
Thank you for your interest for the project. 🤓

Before answering your questions, I want to let you know more about the context around the version 2.1 in order for you to understand better:
We deployed this version few weeks ago, and the current “master” branch is a little messy and experimental because we are trying to deploy it online as fast as possible...
Anyway I will definitely improve the architecture, clean the code and add more comments ASAP.

So now let’s try to answer your questions:

**1. Why don't you use one point to enter all yours application?
2. You copy-paste a lot of duplicate things, with same functionality...**
It will be definitely better to have only one entry point and I'm gonna do that in my next commit.
Basically in the version 2.0 we had only one page (the homepage) then we added the company page and the admin sections. so I just copied the controller for all the pages and changed the code for each. 😖

**3. For company info from Instagram, why not try to use some API. Instagram already have it, so...**
If you are talking about The GraphQL API publicly available (without token) we already use it to fetch data from Instagram. I knew that instead of calling the normal Instagram URL (https://www.instagram.com/wuwana.es) and extract the JSON part from the HTML page we can directly ask the JSON by adding "/?__a=1" at the end of the URL. But I don't know if this kind of URL is more restrictive than the normal one. (in terms of maximum number of call from the same IP address)

**4. Well for some reason you use a lot of trigger_error, so again, why not use some Exception Handler for app.**
99% of the trigger_error you saw in the code was to debug some part of the WebApp on the production server. In order to have an output in the PHP logs.
So it will be removed in my next commit.

Thanks again for your comment and happy new year! 🥳

**Comment by Nils85** [comment]
Hey @levogirar,
Now we are working on the version 2.2 can we delete this conversation?
Because, back then, it was about the freshly released version 2.0 but none of this problems apply anymore.
So what’s the point of keeping this discussion open?

**Reply by levogirar** [reply]
I don't want to delete this discussion, but let me see if I could archive it.
